enterprise/server/workflow/config/config.go

```diff
@@ -190,10 +190,51 @@ fi`, dirName, downloadURL)
 }
 
 func buildWithKythe(dirName string) string {
+	// TODO(jdelfino): This script doesn't pass any extra flags to Bazel, beyond those needed to
+	// enable Kythe. This means the build will fail or be invalid if the normal build workflow
+	// passes any important flags. While passing flags on the command line is discouraged,
+	// we'll need to handle this eventually.
 	bazelConfigFlags := `--config=buildbuddy_bes_backend --config=buildbuddy_bes_results_url`
 	return fmt.Sprintf(`
+BZL_MAJOR_VERSION=$(bazel --version | cut -d' ' -f2 | xargs | cut -d'.' -f1)
+
+if [ $BZL_MAJOR_VERSION -lt 7 ]; then
+    BZLMOD_DEFAULT=0
+else
+    BZLMOD_DEFAULT=1
+fi
+
+# starlark-semantics will print out enable_bzlmod if it differs from the default.
+bazel info starlark-semantics > /dev/null 2>&1 | grep "enable_bzlmod" > /dev/null 2>&1
+if [ $? -eq 1 ]; then
+    BZLMOD_ENABLED=$BZLMOD_DEFAULT
+else
+    BZLMOD_ENABLED=$(( 1 - $BZLMOD_DEFAULT ))
+fi
+
 export KYTHE_DIR="$BUILDBUDDY_CI_RUNNER_ROOT_DIR"/%s
-bazel --bazelrc="$KYTHE_DIR"/extractors.bazelrc build --override_repository kythe_release="$KYTHE_DIR" %s //...`, dirName, bazelConfigFlags)
+
+if [ "$BZLMOD_ENABLED" -eq - ]; then
+    # with bzlmod enabled, override_repository will not work unless the repository is already defined
+	# inject_repository will work, but was added in Bazel 8, so we need to handle <8 by
+	# manually adding to MODULE.bazel.
+    if [ $BZL_MAJOR_VERSION -lt 8 ]; then
+        echo "Adding kythe repository to MODULE.bazel"
+        echo >> MODULE.bazel
+		echo 'bazel_dep(name = "kythe", version = "0.0.75")' >> MODULE.bazel
```

Comment:
nit: indentation looks a bit off, and `echo >> MODULE.bazel` looks redundant

```suggestion
  if [ $BZL_MAJOR_VERSION -lt 8 ]; then
    echo "Adding kythe repository to MODULE.bazel"
		echo 'bazel_dep(name = "kythe", version = "0.0.75")' >> MODULE.bazel
```